src/main/java/com/_up/megastore/services/implementations/SizeService.java

```diff
@@ -41,4 +41,24 @@ public SizeResponse updateSize(UUID sizeId,UpdateSizeRequest updateSizeRequest){
 
         return SizeMapper.toSizeResponse(sizeRepository.save(size));
     }
+
+    @Override
+    public SizeResponse deleteSize(UUID sizeId){
+        ifSizeExistThrowException(sizeId);
+        Size size = ifSizeIsNotDeletedThrowException(sizeId);
+        size.setDeleted(true);
+
+        return SizeMapper.toSizeResponse(sizeRepository.save(size));
+    }
+
+    public void ifSizeExistThrowException(UUID sizeId){
+        if(!sizeRepository.existsById(sizeId)){
+            throw new NoSuchElementException("Size with id " + sizeId + " does not exist.");
+        }
+    }
+
+    public Size ifSizeIsNotDeletedThrowException(UUID sizeId){
+        return sizeRepository.findBySizeIdAndDeletedIsFalse(sizeId)
+                .orElseThrow(() -> new NoSuchElementException("Size with id " + sizeId + " is deleted."));
+    }
```

Comment:
Lo mismo con este metodo, ya que no solo estas verificando si el size no está eliminado, sino que ademas estás retornando el size, seria más algo como esto

```suggestion
    public Size getSizeByIdAndDeletedFalseOrThrowException(UUID sizeId){
        return sizeRepository.findBySizeIdAndDeletedIsFalse(sizeId)
                .orElseThrow(() -> new NoSuchElementException("Size with id " + sizeId + " is deleted."));
    }
```